ngraph/core/src/function.cpp

```diff
@@ -640,3 +647,66 @@ ov::Output<ov::Node> ov::Function::input(const std::string& tensor_name) {
     }
     throw ov::Exception("Input for tensor name " + tensor_name + " was not found.");
 }
+
+void ov::Function::reshape(const std::map<std::string, ov::PartialShape>& partial_shapes) {
+    if (partial_shapes.empty())
+        return;
+
+    const auto& params = get_parameters();
+    std::unordered_map<std::string, std::shared_ptr<ov::op::v0::Parameter>> tensor_param_map;
+
+    // Check that we need to do reshape only if input shapes will be changed
+    bool need_reshape = false;
+    for (const auto& partial_shape : partial_shapes) {
+        bool shape_is_used = false;
+
+        for (const auto& param : params) {
```

Comment:
Should we check that map doesn't contain different shapes for one parameter with several tensor names?

Reply:
yes. In order to prevents cases when user provided map with operation names are they are just ignored